draft-kazuho-httpbis-priority.md

```diff
@@ -112,33 +112,93 @@ Unknown parameters MUST be ignored.
+|              -1 | blocking ({{blocking}})         |
+|               0 | document ({{document}})         |
+| between 1 and 5 | non-blocking ({{non-blocking}}) |
+|               6 | background ({{background}})     |
+{: #urgencies title="Urgencies"}
 
-The default value is `document`.
+The value is encoded as an sh-integer.  The default value is zero.
 
-A server SHOULD transmit HTTP responses in the order of their urgency:
-`blocking` first, followed by `document`, followed by `non-blocking`.
+A server SHOULD transmit HTTP responses in the order of their urgency values.
+The lesser the value is, the higher the precedence is.
 
 The following example shows a request for a CSS file with the urgency set to
-`blocking`:
+`-1`:
 
 ~~~ example
 :method = GET
 :scheme = https
 :authority = example.net
 :path = /style.css
-priority = urgency=blocking
+priority = urgency=-1
 ~~~
 
+The definition of the urgencies and their expected use-case are described below.
+Endpoints SHOULD respect the definition of the values when assigning urgencies.
+
+### blocking
+
+The blocking urgency indicates that the response prevents other responses from
+being used.
+
+Value of -1 is used to represent this urgency.
+
+For example, a stylesheet being referred from the HEAD section of an HTML
+document blocks a web browser from rendering the HTML.  In such case, the
+stylesheet is given the blocking urgency.
+
+### document
+
+The document urgency indicates that the response contains the document that is
+being processed.  This urgency is also used for responses that deserve the same
+precedence as the contents of the document.
+
+Value of zero (0) is used to represent this urgency.
+
+For example, when a user using a web browser navigates to a different HTML
+document, the request for that HTML is given the document urgency.  When that
+HTML document uses a custom font, the request for that custom font SHOULD also
+be given the document urgency.  This is because the availablity of the custom
+font is the precondition for the user to use the portion of the HTML document,
+which is to be rendered by that font.
+
+### non-blocking
+
+The non-blocking urgency indicates that the response does not prevent the client
+from using the document even though the response is being incorporated to or
+referred to by the document.
+
+For example, inline images typically improve the usability of an HTML document.
+Therefore, such images will be given the non-blocking urgency.
+
+Values between 1 and 5 are used to represent this urgency, to provide wiggle
+room for endpoints to give some responses more or less precedence than others
+that belong to the non-blocking group.
+
+Clients SHOULD NOT use values 1 and 5.  Servers MAY use these values to
+prioritize a response above or below other non-blocking responses.
+
+Clients MAY use values 2 and 4 to indicate that a request is given relatively
+higher or lower priority within the non-blocking urgency group.
```

Comment:
if you agree with the comment about the "inline images" above, then here you might add something like:

For example, an image certain to be visible at the top of the page, might be assigned a value of 4 instead of 3, as it will have a high visual impact for the user. 

And if you're feeling verbose:
Conversely, smaller images such as icons or images lower on the page that have less immediate visual impact can be assigned an urgency of value 2. 

To complete the cycle of image-based examples:
Finally, if there is a single, large and important image known to be on the top of the page (sometimes called a hero image), the server can assign it a value of 5 to make sure it is shown to the user as soon as possible.

Reply:
Thank you for the suggestions.

In f73b3cf, I've taken the first sentence for talking about urgency=2, and used async JavaScript file as an example for urgency=4. 49ea38b adds reference to #merging in respect to urgencies 1 and 5.

Reply:
Added feedback inline in the first commit, second one looks good!